src/wasm/wasm-binary.cpp

```diff
@@ -590,7 +590,14 @@ void WasmBinaryWriter::writeElementSegments() {
     bool hasTableIndex = false;
     if (!isPassive) {
       tableIdx = getTableIndex(segment->table);
-      hasTableIndex = tableIdx > 0;
+      // Table index 0 is special for backwards compatibility: only tables with
```

Comment:
Maybe something like this would be clearer: `// The table index can and should be elided for active segments of table 0 when table 0 has type funcref. This was the only type of segment supported by the MVP, which also did not support table indices in the segment encoding.`

Reply:
Thanks, added, that does read better.